### PR TITLE
feature #7

### PR DESCRIPTION
**Pull request by arnochauveau**
implemented #7

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -49,6 +50,8 @@ var CONFIG = {
     }
 };
 
+var configFilePath = __dirname + "/.git-issues-config.json";
```
Comment (IonicaBizau) [inline index.js]: Could you please store this file in the home directory? You can use the [`ul`](https://github.com/IonicaBizau/node-ul) package and do:

```js
var ul = require("ul");
...
var configFilePath = ul.USER_DIR + "/.git-issues-config.json";
```

File: index.js [hunk 2]
```diff
@@ -76,28 +79,40 @@ if (['github', 'bitbucket'].indexOf(son.source) === -1) {
     process.exit(11);
 }
 
-// if a user is provided, we also need a password for the basic authentication
-if (user && !argv['password']) {
+//if json config file exists, use its credentials.
 
-    prompt.get(CONFIG.promptSchema, function(err, result) {
+if (fs.existsSync(configFilePath)) {
```
Comment (IonicaBizau) [inline index.js]: Use [`is-there`](https://github.com/IonicaBizau/node-is-there) here since `exists*` will be deprecated.

File: index.js [hunk 3]
```diff
@@ -138,10 +153,11 @@ function getIssues(son, user, pass, callback) {
     } catch (err) {}
 
     if (!provider) {
-        return callback('Provider not suppoerted: ' + son.source);
+        return callback('Provider not supported: ' + son.source);
```
Comment (IonicaBizau) [inline index.js]: :+1:

**Comment by IonicaBizau** [comment]
Thanks! I left some comments. This is awesome! :sparkles:

**Comment by arnochauveau** [comment]
no problemo :) i'll make the changes

File: package.json [hunk 4]
```diff
@@ -16,9 +16,15 @@
...
+  "author": {
+    "name": "Gabriel Petrovay",
+    "email": "gabipetrovay@gmail.com"
+  },
   "contributors": [
-    "Ionică Bizău <bizauionica@gmail.com>"
+    {
+      "name": "Ionică Bizău",
+      "email": "bizauionica@gmail.com"
+    }
```
Comment (IonicaBizau) [inline package.json]: If you like, add yourself as contributor. :smile:

File: package.json [hunk 5]
```diff
@@ -34,7 +40,35 @@
     "prompt": "0.2.14",
     "optimist": "0.6.1",
     "request": "2.55.0",
-    "giturlparse": "1.2.0"
+    "giturlparse": "1.2.0",
+    "ul":"2.10.1", 
+    "is-there" : "2.10.1"
+  },
+  "devDependencies": {},
+  "gitHead": "d73b98a25ce7a7f47ca71c3b13e5a49d3bdb97a2",
```
Comment (IonicaBizau) [inline package.json]: Can you please clean this up? These are added automatically by npm when installing a module, I'm not sure how they appeared here.

**Comment by arnochauveau** [comment]
:D i will, thanks for the tips btw! didn't know fs.exists will be deprecated!

File: package.json (hunk 5 above)
Comment (arnochauveau) [inline package.json]: isn't that the point? I guess no one has the ul and is-there packages installed? :)

**Comment by arnochauveau** [comment]
@IonicaBizau isn't that the point? Assuming no one has the ul and is-there packages installed? :)

**Comment by IonicaBizau** [comment]
@arnochauveau I'm not sure what you mean. The `ul` and  `is-there` packages will be installed automatically since they are dependencies.

**Comment by arnochauveau** [comment]
@IonicaBizau yeah I was a bit confused I guess, The packages in node_modules get installed automatically so I don't have to include them in the package.json

File: index.js [hunk 6]
```diff
@@ -155,16 +172,18 @@ function getIssues(son, user, pass, callback) {
...
     }
+    
 
     // fetch the issues from the web
     request.get(options, function(err, response, issueRespose) {
-
+       
         if (err) {
             return callback(err);
         }
```
Comment (IonicaBizau) [inline index.js]: Somewhere in line 194, would be nice to stringify (`JSON.stringify`) the `options.auth` object and write create the config file.

**Comment by arnochauveau** [comment]
Done :)

**Comment by IonicaBizau** [comment]
Cool! I will check it these days and merge it! Thanks! :sparkles:

**Comment by arnochauveau** [comment]
:+1:

**Pull request merged**